### ISSUE TITLE
Scrolling lags when adding large number of items in section

### ISSUE THREAD

**Issue by aayushthakur** [lisawray/groupie#161]
In the data binding example, when I add more items to the section recyclerview instead of 12 say 500, scrolling starts to lag. This doesn't happen in the infinite loading section.

![scroll-lag](https://user-images.githubusercontent.com/16386061/35570186-7a085404-05f4-11e8-9ce9-253ccb6a2479.gif)

**Comment by Revell1on** [comment]
As far as i have found by now - the main problem with this lag is spanSizeLookup and all the iterations caused by GroupAdapter#getItem called inside it.
adapter has to iterate though all items to find out every item span size.
Also the problem appears only when scrolling from bottom to top

**Comment by aayushthakur** [comment]
Yes I know that the lag only occurs when scrolling bottom to top, in my usage I can have a large number of items inside the section even more than 500 and further more the items are images.

**Comment by lisawray** [comment]
Better lookups are definitely the key here.  I've started work on a caching structure for lookups but haven't finished yet.  Would love it if anyone wants to help!
